KAS-3572: Split up Case and Decisionmaking-flow — migrations

sergiofenoll opened the pull request:
PR consists of 4 migrations, one for each graph, that create a new `decisionmaking-flow` for each `case` that is encountered. Only cases that are linked to a Ministerraad are considered here.

Writing the migrations as a single query (using `OPTIONAL` clauses for the properties of `case` that were being copied) gave weird results, so I opted to split it up into a number of queries. There is also one migration per graph because writing the migration to act on all graphs at once (using a variable for the graph) also gave incorrect results.

MikiDi committed:
split dossier & besluitvormingsaangelegenheid.  resources config

sergiofenoll committed:
feat: add migrations to split up cases and decisionmaking-flows

Four migrations, one for each graph, because writing the migration to
work over all graphs (with a variable) gave weird results.

MikiDi committed:
update search config for case/flow split

MikiDi committed:
remove obsolete mandatee full name path from search

MikiDi committed:
Merge branch 'development' into feature/KAS-2326-aangelegenheid-data-model

sergiofenoll committed:
fix: insert UUID triple as well

sergiofenoll committed:
fix: remove trailing commas in search config

sergiofenoll committed:
fix: re-add short-title to case and document that the dmf props are unused

sergiofenoll committed:
refactor: rename case to decisionmaking-flow in relation of subcase

sergiofenoll commented:
Adding the changes from https://github.com/kanselarij-vlaanderen/app-kaleidos/tree/feature/KAS-2326-aangelegenheid-data-model to this PR (otherwise getting it running on Jenkins is going to get tricky)

sergiofenoll committed:
Merge branch 'feature/KAS-2326-aangelegenheid-data-model' into feature/KAS-3572-case-decisionmaking-flow-migrations

sergiofenoll committed:
Merge branch 'development' into feature/KAS-3572-case-decisionmaking-flow-migrations

sergiofenoll committed:
fix: refine query to match MR cases

We need to match both those cases that don't have a publication-flow,
i.e. cases created via the "Dossiers" tab that haven't been published
yet via an agenda. For the cases that do have a publication-flow, we
only want to match those that are related to an agenda. Before we only
matched the latter, which means we missed out on a ton of cases.

sergiofenoll committed:
chore: update testdata.zip

sergiofenoll committed:
ci: use PR tagged builds of services

erikap commented on config/migrations/20220825163000-case-and-decisionmaking-flow-migration-kanselarij.sparql:
```diff
@@ -0,0 +1,106 @@
+    }
+  }
+
+  BIND(STRUUID() AS ?uuid)
```
Genereert dit effectief verschillende UUIDs per instance? Dit lijkt mij een case zoals https://github.com/openlink/virtuoso-opensource/issues/515. Workaround is voorzien in https://github.com/openlink/virtuoso-opensource/issues/515#issuecomment-456848368

sergiofenoll replied:
Double checked to be sure, seems like it does generate different UUIDs.
![image](https://user-images.githubusercontent.com/22411874/188720957-b102bc82-fb4a-4fb6-8975-149adba0661a.png)

Each migration took about ~40 seconds to run on a copy of the ACC database, for reference.

erikap reviewed (changes requested):
Huidige migrations gaan verschillende UUIDs/URIs genereren voor nieuwe besluitvormingsaangelegenheden in de verschillende graphs. Deze moeten overeenstemmen over de graphs heen.

Ofwel moeten de migrations voor minister/overheid/regering aangepast worden zodanig dat ze de nieuwe besluitvormingsaangelegenheden uit de kanselarij-graph kopiëren; ofwel wordt enkel een migration uitgevoerd op de kanselarij-graph en laten we Yggdrasil een volledige propagatie doen bij de deploy.

Is de migration getest op een DB met grootteorde van prod?

erikap commented on config/resources/dossier-domain.lisp:
```diff
@@ -40,10 +56,11 @@
-                (:modified            :datetime ,(s-prefix "ext:modified")))
-  :has-one `((case                    :via ,(s-prefix "dossier:doorloopt")
+                (:modified            :datetime ,(s-prefix "ext:modified"))
+                (:show-as-remark      :boolean ,(s-prefix "ext:wordtGetoondAlsMededeling"))) ;; TODO: Is show-as-remark still required here? Might have come back in through merge.
```
Lijkt me idd overbodig

erikap commented on config/search/config.json:
```diff
@@ -201,19 +201,25 @@
     {
-      "type": "case",
+      "type": "decisionmaking-flow",
       "on_path": "cases",
```
Specifieke reden waarom het path hier niet mee aangepast is? Zou dat wel doen om verwarring te vermijden.

sergiofenoll replied:
Niet echt, was gewoon al zo. Weet niet of Michaël daar een specifieke reden voor had, of dat gewoon was vergeten aan te passen.

sergiofenoll committed:
chore: remove show-as-remark that creeped back in

sergiofenoll committed:
fix: update case-decisionmaking-flow split migrations

Decisionmaking-flows in other graphs need to have the same URI and UUID
as the ones in kanselarij, so the migrations now copy the resources from
kanselarij instead of trying to make new ones.

sergiofenoll committed:
chore: update path of decisionmaking-flows search

erikap committed:
Remove outdated mapping from search config

sergiofenoll committed:
feat: add government-areas to publication-flow

sergiofenoll committed:
Merge branch 'feature/KAS-3572-case-decisionmaking-flow-migrations' of github.com:kanselarij-vlaanderen/app-kaleidos into feature/KAS-3572-case-decisionmaking-flow-migrations

sergiofenoll committed:
Merge branch 'development' into feature/KAS-3572-case-decisionmaking-flow-migrations

erikap committed:
Merge branch 'development' into feature/KAS-3572-case-decisionmaking-flow-migrations

ValenberghsSven committed:
KAS-3572 migrations done on testdb and yggdrasil rebuild

erikap committed:
Use tagged versions for yggdrasil, case- and gov-field sync services
